### DISCUSSION TITLE
Textfields should default to text type, not string

### DISCUSSION THREAD

**Discussion by braco** [directus/directus#18760] [Feature Requests]
### Summary

I've already been burned a few times since there's no way to change the field type once it's been created.

The default for textfields is "string", which has a small size limit. I'm assuming this is using varchar behind the scenes?

In Postgres, there's not much reason to use varchar over text, it's mostly just a footgun:

https://maximorlov.com/char-varchar-text-postgresql/

### Motivation

I've been using Directus for a few days and have already needed to completely wipe Collections and start over as I was importing data.

### Detailed Design

For text inputs, use more forgiving "text" type behind the scenes and set a soft limit for length

### Must Have:
- Change default type to text

### Drawbacks

None?

### Alternatives

Allow changing field types after they've been created

### Adoption Strategy

Nothing needed

**Comment by bamboowonder** [comment]
You can change the varchar256 to just plain VARCHAR (without a limiting number) or TEXT directly in the database and directus will reflect that change in the app.
And YES, I could not agree more!!  I'd rather be using citext in the database, but there is a number of features that aren't available unless using varchar, but then you lose case-insensitivity...   It's been a rough transition in that regard with Directus.

**Reply by braco** [reply]
What features are disabled without varchar?

**Reply by bamboowonder** [reply]
Directus tries to be clever making assumptions about what field types get certain features.   Unfortunately it's wrong in our case.
Try making a varchar field.  and make a text field.  and note the quite different available options in the Interface and Display area.
As a result we've had to do a lot of shoehorning of our database to work with Directus.

**Comment by rijkvanzanten** [comment]
> In Postgres, there's not much reason to use varchar over text, it's mostly just a footgun:

_In Postgres_ is the key part of the phrase here isn't it 🙂 We'll have to do some research to make sure that applies to all database vendors before we just change it to `text` everywhere.

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
